atom.sh

```diff
@@ -148,6 +148,15 @@ trap 'on_die' SIGQUIT SIGTERM
 
 # If the wait flag is set, don't exit this process until Atom tells it to.
 if [ $WAIT ]; then
+  WAIT_FIFO="$ATOM_HOME/.wait_fifo"
+  while true; do
+    [ -f "$WAIT_FIFO" ] && rm "$WAIT_FIFO"
+    [ ! -p "$WAIT_FIFO" ] && mkfifo "$WAIT_FIFO"
+    read < "$WAIT_FIFO" || break
+    sleep 1 # prevent a tight loop
+  done
+  
+  # fall back to sleep
```

Comment:
Can you explain why this second "fall back" while loop is still needed?

If the `read` completes (presumably due to an error), won't we `sleep` and then recreate the pipe? I would think that we would already "fall back" to sleep, even without the second `while` loop.

Reply:
If we (for some reason) can't create the fifo or remove the file that's there in its place, I guess.

If you want, I can remove it, though.

Reply:
My knowledge of bash is somewhat limited, so I might be missing something.

I would think that it'd be impossible to reach that code though, because the `while` loop above would never complete. Is that true?

Reply:
there's a break after the `read`

Reply:
Oh, thanks for explaining; I didn't notice that.

So why is the `read` wrapped in a `while` loop? In what circumstance would `read` exit successfully?

Reply:
For example if you write to the FIFO.
I don't know why you would, again, but it's possible.

Reply:
I see, so in that case it should be fine to `read` again rather than falling back to a `sleep`, but we don't want to retry if there appears to be a problem with the fifo. That actually makes perfect sense.